src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -1,88 +1,208 @@
+#ifndef __NBL_C_VULKAN_PHYSICAL_DEVICE_H_INCLUDED__
+#define __NBL_C_VULKAN_PHYSICAL_DEVICE_H_INCLUDED__
+
 #include "nbl/video/IPhysicalDevice.h"
 
 #include <volk.h>
 
 #include "nbl/video/CVKLogicalDevice.h"
+#include "nbl/video/surface/ISurfaceVK.h"
 
-namespace nbl {
-namespace video
+namespace nbl::video
 {
-
+        
 class CVulkanPhysicalDevice final : public IPhysicalDevice
 {
 public:
-    CVulkanPhysicalDevice(VkPhysicalDevice _vkphd) : m_vkphysdev(_vkphd) 
+    CVulkanPhysicalDevice(VkPhysicalDevice _vkphd, core::smart_refctd_ptr<system::ISystem>&& sys, core::smart_refctd_ptr<asset::IGLSLCompiler>&& glslc)
+        : IPhysicalDevice(std::move(sys), std::move(glslc)), m_vkphysdev(_vkphd)
     {
-        VkPhysicalDeviceProperties props;
-        vkGetPhysicalDeviceProperties(m_vkphysdev, &props);
-        // TODO fill m_properties
-
-        VkPhysicalDeviceFeatures features;
-        vkGetPhysicalDeviceFeatures(m_vkphysdev, &features);
-        // TODO fill m_features
-
+        // Get physical device's limits
+        VkPhysicalDeviceSubgroupProperties subgroupProperties = { VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_PROPERTIES };
+        {
+            VkPhysicalDeviceProperties2 deviceProperties = { VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2 };
+            deviceProperties.pNext = &subgroupProperties;
+            vkGetPhysicalDeviceProperties2(m_vkphysdev, &deviceProperties);
```

Comment:
`vkGetPhysicalDeviceProperties2` is a Vulkan 1.1 thing or 1.2?

Reply:
1.1